src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java

```diff
@@ -0,0 +1,386 @@
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.ibm.mq.constants.CMQC.*;
+
+/**
+ * Add messages to queue
+ */
+public class MQPublishQueue extends AbstractConnector {
+
+    MQConnectionBuilder connectionBuilder;
+    MQConfiguration config;
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        try {
+
+            log.info("Message received at ibm-mq connector");
+
+            //retrieve message body
+            SOAPEnvelope soapEnvelope = messageContext.getEnvelope();
+            OMElement getElement = soapEnvelope.getBody().getFirstElement();
+            String queueMessage = "";
+
+            if (getElement != null) {
+                queueMessage = getElement.toString();
+            } else {
+                queueMessage = "";
+            }
+
+            //access queue manager
+            if(connectionBuilder==null || connectionBuilder.getManager()==null) {
+                connectionBuilder = new MQConnectionBuilder(messageContext);
+                config = connectionBuilder.getConfig();
+            }
+
+            MQQueue queue = setQueue(connectionBuilder, config);
+            int messageType = config.getMessageType();
+            MQMessage mqMessage;
+
+            switch (messageType) {
+                case 1:
+                    mqMessage = buildRequestMessageandGetResponse(config, queueMessage);
+                    break;
+                case 4:
+                    mqMessage = buildReportMessage(config, queueMessage);
+                    break;
+                default:
+                    mqMessage = buildDatagramOrReplyMessage(config, queueMessage);
+            }
+
+            if (queue == null || mqMessage == null) {
+                log.error("Error in publishing message");
+            } else {
+                log.info("queue initialized");
+                queue.put(mqMessage);
+                queue.close();
+                log.info("Message successfully placed at " + config.getQueue() + " queue and closed");
+            }
+
+        } catch (Exception e) {
+            log.error("Problem in publishing to message" + e);
+            throw new ConnectException(e);
+        }
+    }
+
+    /**
+     * Initialize queue
+     */
+    MQQueue setQueue(MQConnectionBuilder connectionBuilder, MQConfiguration config) {
+        MQQueueManager manager = connectionBuilder.getManager();
+        MQQueue queue;
+        try {
+            queue = manager.accessQueue(config.getQueue(), CMQC.MQOO_OUTPUT);
+        } catch (MQException e) {
+            log.error("Error creating queue " + e);
+            return null;
```

Comment:
Can we continue the flow getting the exception from here? Is this null case handled properly at higher layer. Or do we need to throw/propagate this to higher layer?